include/range/v3/view_adaptor.hpp

```diff
@@ -179,6 +179,38 @@ namespace ranges
             {
                 return first();
             }
+
+        protected:
+            // Adaptor accessor
+            Adapt& adaptor()
+            {
+                return second();
+            }
+            const Adapt& adaptor() const
+            {
+                return second();
+            }
+        };
+
+        namespace detail {
+            template<typename BaseSent, typename Adapt>
+            meta::id<base_adaptor_sentinel<BaseSent, Adapt>> base_adaptor_sentinel_2_(long);
+
+            template<typename BaseSent, typename Adapt>
+            meta::id<typename Adapt::template mixin<base_adaptor_sentinel<BaseSent, Adapt>>> base_adaptor_sentinel_2_(int);
+
+            template<typename BaseSent, typename Adapt>
+            struct base_adaptor_sentinel_
+                : decltype(base_adaptor_sentinel_2_<BaseSent, Adapt>(42))
```

Comment:
```suggestion
              : decltype(base_adaptor_sentinel_2_<BaseSent, Adapt>(42))
```

Reply:
Context dependent indent? 3 spaces, not 4, in class parents list?